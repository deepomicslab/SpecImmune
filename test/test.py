import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import sys
import os

def plot_depth(file_path):
    # Check if the file exists
    if not os.path.isfile(file_path):
        print(f"Error: The file {file_path} does not exist.")
        return

    # Load depth data
    depth_data = pd.read_csv(file_path, sep='\t', header=None, names=['chromosome', 'position', 'depth'])

    # Plot using seaborn
    plt.figure(figsize=(12, 6))
    sns.lineplot(data=depth_data, x='position', y='depth', hue='chromosome', linewidth=1)
    # plot a horizontal line at y=30
    plt.axhline(y=24.5, color='r', linestyle='--', label='Threshold (30)')


    plt.title('Depth across multiple chromosomes')
    plt.xlabel('Position')
    plt.ylabel('Depth')
    plt.legend(title='Chromosome')
    plt.show()

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python plot_depth.py <path_to_depth_file>")
    else:
        file_path = sys.argv[1]
        plot_depth(file_path)