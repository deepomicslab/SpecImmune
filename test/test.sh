db=../db/

python ../scripts/main.py -r test.fastq.gz -j 15 -i HLA -n test_HLA -o test --db $db


python3 ../evaluation/assess_typing.py -i HLA --true test.HLA.hap.alleles.txt --infer test/test_HLA/test_HLA.HLA.type.result.txt 
