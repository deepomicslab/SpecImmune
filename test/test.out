python ../scripts/main.py -r CYP/HG03579.CYP.fastq.0.1.fq.gz -j 15 -i CYP -n test_CYP -o test -y pacbio --db ../db --hg38 /home/shuaiw/borg/hg38/hg38_no_alt.fa --db /home/shuaiw/methylation/data/spec/dpython3 ../scripts/main.py --hg38 /home/shuaiw/methylation/data/spec/db -n test_IG_TR -o test -j 10 -y nanopore -i IG_TR -r IG_TR/vdj.fq.gz
python ../scripts/main.py -r KIR/HG00131.KIR.fastq.gz -j 15 -i KIR -n test_KIR -o test --hete_p 0.2 --align_method_1 minimap2 -y pacbio --db /home/shuaiw/methylation/data/spec/db 
